From 2505ca03c0f8c8d36e280f596799e85961029b9f Mon Sep 17 00:00:00 2001
From: Mathias Kraus <elboberido@m-hias.de>
Date: Wed, 13 Aug 2025 15:28:04 +0200
Subject: [PATCH] Fix iceoryx hoofs fetching

---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c5d0c998..c10a2952 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,7 +13,6 @@
 cmake_minimum_required(VERSION 3.22)
 
 set(ICEORYX_HOOFS_VERSION 2.95.6)
-include(${CMAKE_CURRENT_LIST_DIR}/cmake/fetch-iceoryx-hoofs.cmake)
 
 set(IOX2_VERSION 0.6.1)
 project(iceoryx2-cxx VERSION ${IOX2_VERSION} LANGUAGES CXX)
-- 
2.50.1

